src/main/java/move/ElephantMovement.java

```diff
@@ -0,0 +1,64 @@
+package move;
+
+import static direction.Direction.DOWN;
+import static direction.Direction.DOWN_LEFT_DIAGONAL;
+import static direction.Direction.DOWN_RIGHT_DIAGONAL;
+import static direction.Direction.LEFT;
+import static direction.Direction.RIGHT;
+import static direction.Direction.UP;
+import static direction.Direction.UP_LEFT_DIAGONAL;
+import static direction.Direction.UP_RIGHT_DIAGONAL;
+
+import direction.Direction;
+import direction.Point;
+import java.util.List;
+import java.util.Map;
+import piece.Pieces;
+
+public class ElephantMovement implements MovementRule {
+
+    private static final Map<Point, List<Direction>> paths = Map.of(
+            new Point(-2, -3), List.of(UP, UP_LEFT_DIAGONAL),
+            new Point(2, -3), List.of(UP, UP_RIGHT_DIAGONAL),
+            new Point(-3, -2), List.of(LEFT, UP_LEFT_DIAGONAL),
+            new Point(-3, 2), List.of(LEFT, DOWN_LEFT_DIAGONAL),
+            new Point(-2, 3), List.of(DOWN, DOWN_LEFT_DIAGONAL),
+            new Point(2, 3), List.of(DOWN, DOWN_RIGHT_DIAGONAL),
+            new Point(3, -2), List.of(RIGHT, UP_RIGHT_DIAGONAL),
+            new Point(3, 2), List.of(RIGHT, DOWN_RIGHT_DIAGONAL)
+    );
+
+    private final int direction;
+
+    public ElephantMovement(int direction) {
+        this.direction = direction;
+    }
+
+    @Override
+    public Point move(Pieces pieces, Point from, Point to) {
+        List<Direction> directions = paths.getOrDefault(to.minus(from), List.of());
+        validateInvalidDestination(directions);
+
+        Point checkPoint = new Point(from.x(), from.y());
+        validateExistPieceInPath(pieces, directions, checkPoint);
+
+        return to;
+    }
+
+    private void validateInvalidDestination(List<Direction> directions) {
+        if(directions.isEmpty()) {
+            throw new IllegalArgumentException("[ERROR] 선택할 수 없는 목적지입니다.");
+        }
+    }
+
+    private void validateExistPieceInPath(Pieces pieces, List<Direction> directions, Point checkPoint) {
+        if (checkExistPieceInPoint(pieces, directions, checkPoint)) {
+            throw new IllegalArgumentException("[ERROR] 경로에 기물이 존재합니다.");
+        }
+    }
+
+    private boolean checkExistPieceInPoint(Pieces pieces, List<Direction> directions, Point checkPoint) {
+        return directions.stream()
+                .anyMatch(direction -> pieces.isExistPieceIn(checkPoint.plus(direction.multiply(this.direction))));
```

Comment:
```suggestion
                .anyMatch(direction -> {
                    var 적절한곱셈변수명 = direction.multiply(this.direction);
                    var 적절한더하기변수명 = checkPoint.plus(적절한곱셈변수명);
                    return pieces.isExistPieceIn();
                })
```

한 줄에 다 작성하기 보다 코드 가독성을 위해 변수로 나눠보세요. 다른 개발자가 코드 파악하는데 도움이 됩니다.

Reply:
해당 피드백을 보고 변수로 나누는 과정에서 로직 자체에 결함이 있음을 발견하여 새롭게 로직을 짰습니다! 감사합니다.

커밋: [84fa671](https://github.com/woowacourse/java-janggi/pull/72/commits/84fa6712a0c362bb428d7d9d3ec1926bb1fbe646)